### PR TITLE
Add a A Small Tour in Introduction page

### LINKED ISSUE #939: There is no page that explains how to navigate between URL, controller and view
There is no page that explains how to navigate between URL, controller and view for Front Office and Back Office.

It can be difficult for newcomers to understand by themselves.

### PR DESCRIPTION
**Pull request by matks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Add a "A Small Tour" page in Introduction page. It tries to explain how to navigate between URL, controller and view for Front Office and Back Office.
| Fixed ticket? | Fixes https://github.com/PrestaShop/docs/issues/939

### CODE REVIEW

File: src/content/1.7/basics/a-small-tour.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+The legacy backend relies on controllers you can find in directory `controllers/admin`
+and PrestaShop classes that contain the business logic, mainly from `classes` folder.
+
+For legacy pages, the views and the Javascript can be found in `admin-dev/themes/default`.
+
+The Symfony backend relies on controller you can find in `src/PrestaShopBundle/Controller/Admin` and PrestaShop logic that mostly come from `src` folder, and also some `classes` files.
+
+For migrated pages, the views can be found in `src/PrestaShopBundle/Resources/views`.
+The Javascript ccan be found in `admin-dev/themes/new-theme/js`.
```
Comment (NeOMakinG) [inline src/content/1.7/basics/a-small-tour.md:47]:
```suggestion
The Javascript can be found in `admin-dev/themes/new-theme/js`.
```

**Comment by matks** [comment]
CI is failing because of https://github.com/PrestaShop/docs/issues/943
https://github.com/PrestaShop/docs/pull/944/ will fix CI

**Review by eternoendless — COMMENTED** [review]
I don't know, I feel that this is at the same time too in-depth to be placed in "Basics" and too vague to be placed in "Core Reference". 

I find that it would be better to update / improve the [introduction](https://devdocs.prestashop.com/1.7/basics/introduction/), then document routing in Core Reference.

File: src/content/1.7/basics/a-small-tour.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+Note that a live demo is availalbe on [demo.prestashop.com](https://demo.prestashop.com/).
+
+## Front office MVC
+
+The Front Office is a small application being powered by PHP and Smarty.
+
+The backend relies on controllers you can find in directory `controllers/front`
+and PrestaShop classes that contain the business logic, mainly from `classes` folder.
+
+The views and the Javascript come from the installed theme that you will find in `themes/`. You can have multiple themes available on a shop but only one is enabled and in use.
```
Comment (zuk3975) [inline src/content/1.7/basics/a-small-tour.md:19]:
```suggestion
The views and the Javascript comes from the installed theme that you will find in `themes/`. You can have multiple themes available on a shop but only one is enabled and in use.
```

File: src/content/1.7/basics/a-small-tour.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+### Visit a Front controller
+
+When you visit `/` on your shop you are being returned the Homepage.
+
+What happened behind the scenes is that
+
+1. Your HTTP request hit the `index.php` main routing file that relies on the _Dispatcher_
+2. The Dispatcher found the right Front controller ; the `IndexController`
```
Comment (zuk3975) [inline src/content/1.7/basics/a-small-tour.md:28]:
```suggestion
2. The Dispatcher found the right Front controller - the `IndexController`
```

File: src/content/1.7/basics/a-small-tour.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+3. The `IndexController` calls the `index` Smarty view to be rendered, populated with PHP objects
+
+If you send another HTTP request, the same sequence will happen
+- Dispatcher finds the right Front controller
+- Front controller builds the right response using a Smarty template and relying on PrestaShop business logic
+
+## Back office MVC
+
+The Back Office is a large application being powered by PHP and Smarty for the legacy part and by PHP, Symfony and Twig for the migrated part.
```
Comment (zuk3975) [inline src/content/1.7/basics/a-small-tour.md:37]: Did we explain somewhere what is the `legacy` and what is `migration`.
In other words, Is the reader aware of the following:
* Prestashop codebase is being migrated to Symfony framework, therefore the keyword "**legacy**" means old, not migrated (and most of the time poor quality) code, while "**migrated**" code is the one that was recently updated following best practices (as much as possible) and advantages of Symfony framework (link to symfony maybe).

File: src/content/1.7/basics/a-small-tour.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+For legacy pages, the views and the Javascript can be found in `admin-dev/themes/default`.
+
+The Symfony backend relies on controller you can find in `src/PrestaShopBundle/Controller/Admin` and PrestaShop logic that mostly come from `src` folder, and also some `classes` files.
+
+For migrated pages, the views can be found in `src/PrestaShopBundle/Resources/views`.
+The Javascript can be found in `admin-dev/themes/new-theme/js`.
+
+### Visit a legacy Back Office controller
+
+When you visit `/admin-{xxx}/index.php?controller=AdminCarriers&token={yyy}` on your shop you are being returned the Carriers Back Office page.
```
Comment (zuk3975) [inline src/content/1.7/basics/a-small-tour.md:51]:
```suggestion
When you visit `/admin{x}/index.php?controller=AdminCarriers&token={y}` on your shop you are being returned the Carriers Back Office page.
```
the `-` is not always there, as I've seen its only admin-dev that has the dash isn't that right?

File: src/content/1.7/basics/a-small-tour.md (hunk 4 above)
Comment (matks) [reply src/content/1.7/basics/a-small-tour.md]: I moved this page inside Introduction where legacy and migrated are defined.

**Comment by matks** [comment]
@eternoendless @zuk3975 I applied your feedbacks.

I added A Small Tour content in Introduction. I will submit another PR about the routing.

**Comment by matks** [comment]
Could you please review this Pull Request @PrestaShop/prestashop-maintainers

File: src/content/1.7/basics/introduction.md [hunk 6]
```diff
@@ -90,3 +90,60 @@ The driving idea is that we want our code to be more robust, more modular, and f
...
+Here is a small guide to help you navigate in the software.
+
+Note that a live demo is available on [demo.prestashop.com](https://demo.prestashop.com/).
+
+### Front office MVC
+
+The Front Office is a small application being powered by PHP and Smarty.
+
+The backend relies on controllers you can find in directory `controllers/front`
+and PrestaShop classes that contain the business logic, mainly from `classes` folder.
```
Comment (kpodemski) [inline src/content/1.7/basics/introduction.md:105]:
```suggestion
and PrestaShop classes that contain the business logic, mainly from the `classes` folder.
```

File: src/content/1.7/basics/introduction.md [hunk 7]
```diff
@@ -90,3 +90,60 @@ The driving idea is that we want our code to be more robust, more modular, and f
...
+
+The views and the Javascript comes from the installed theme that you will find in `themes/`. You can have multiple themes available on a shop but only one is enabled and in use.
```
Comment (kpodemski) [inline src/content/1.7/basics/introduction.md:107]:
```suggestion
The views and the Javascript come from the installed theme that you will find in `themes/`. You can have multiple themes available on a shop but only one is enabled and in use.
```

File: src/content/1.7/basics/introduction.md [hunk 8]
```diff
@@ -90,3 +90,60 @@ The driving idea is that we want our code to be more robust, more modular, and f
...
+
+### Visit a Front controller
+
+When you visit `/` on your shop you are being returned the Homepage.
+
+The HTTP request you sent was received by the _Dispatcher_ which found the right Front controller (the `IndexController` for Homepage).
+
+This `IndexController` returned a HTTP response containing a HTML document rendered by Smarty.
```
Comment (kpodemski) [inline src/content/1.7/basics/introduction.md:115]:
```suggestion
This `IndexController` returned an HTTP response containing an HTML document rendered by Smarty.
```

we prefer `aitch` or `haitch`? 😅

File: src/content/1.7/basics/introduction.md [hunk 9]
```diff
@@ -90,3 +90,60 @@ The driving idea is that we want our code to be more robust, more modular, and f
...
+
+## Back office MVC
+
+The Back Office is a large application being powered by PHP and Smarty for the legacy part and by PHP, Symfony and Twig for the migrated part.
+
+The legacy backend relies on controllers which you can find in directory `controllers/admin`. It also relies on PrestaShop classes that contain the business logic, mainly from `classes` folder.
```
Comment (kpodemski) [inline src/content/1.7/basics/introduction.md:121]:
```suggestion
The legacy backend relies on controllers which you can find in directory `controllers/admin`. It also relies on PrestaShop classes that contain the business logic, mainly from the `classes` folder.
```

File: src/content/1.7/basics/introduction.md [hunk 10]
```diff
@@ -90,3 +90,60 @@ The driving idea is that we want our code to be more robust, more modular, and f
...
+
+For legacy pages, the views and the Javascript can be found in `admin-dev/themes/default`.
+
+The Symfony backend relies on controller you can find in `src/PrestaShopBundle/Controller/Admin` and PrestaShop logic that mostly come from `src` folder, and also some `classes` files.
```
Comment (kpodemski) [inline src/content/1.7/basics/introduction.md:125]:
```suggestion
The Symfony backend relies on the controller you can find in `src/PrestaShopBundle/Controller/Admin` and PrestaShop logic that mostly comes from the `src` folder, and also some `classes` files.
```

File: src/content/1.7/basics/introduction.md [hunk 11]
```diff
@@ -90,3 +90,60 @@ The driving idea is that we want our code to be more robust, more modular, and f
...
+
+For migrated pages, the views can be found in `src/PrestaShopBundle/Resources/views`.
+The Javascript can be found in `admin-dev/themes/new-theme/js`.
+
+### Visit a legacy Back Office controller
+
+When you visit `/admin{x}/index.php?controller=AdminCarriers&token={y}` on your shop you are being returned the Carriers Back Office page.
```
Comment (kpodemski) [inline src/content/1.7/basics/introduction.md:132]:
```suggestion
When you visit `/admin{x}/index.php?controller=AdminCarriers&token={y}` on your shop you are being returned to the Carriers Back Office page.
```

File: src/content/1.7/basics/introduction.md [hunk 12]
```diff
@@ -90,3 +90,60 @@ The driving idea is that we want our code to be more robust, more modular, and f
...
+
+The HTTP request you sent was received by the Back Office _Dispatcher_ which found the right Admin controller (the `AdminCarriersController`).
+
+This `AdminCarriersController` returned a HTTP response containing a HTML document rendered by Smarty.
```
Comment (kpodemski) [inline src/content/1.7/basics/introduction.md:136]:
```suggestion
This `AdminCarriersController` returned an HTTP response containing an HTML document rendered by Smarty.
```

File: src/content/1.7/basics/introduction.md [hunk 13]
```diff
@@ -90,3 +90,60 @@ The driving idea is that we want our code to be more robust, more modular, and f
...
+
+The Admin controllers use a generic system to choose what Smarty template use. Generic templates to display forms and listings are available and the controller provides the structure configuration (for example it controls what columns and rows are displayed).
+
+### Visit a migrated Back Office controller
+
+When you visit `/admin-{xxx}/index.php/configure/shop/preferences/preferences?_token={yyy}` on your shop you are being returned the Preferences Back Office page.
```
Comment (kpodemski) [inline src/content/1.7/basics/introduction.md:142]:
```suggestion
When you visit `/admin-{xxx}/index.php/configure/shop/preferences/preferences?_token={yyy}` on your shop you are being returned to the Preferences Back Office page.
```

File: src/content/1.7/basics/introduction.md [hunk 14]
```diff
@@ -90,3 +90,60 @@ The driving idea is that we want our code to be more robust, more modular, and f
...
+
+The HTTP request you sent was received by the Symfony the kernel that handled your request.
```
Comment (kpodemski) [inline src/content/1.7/basics/introduction.md:144]:
```suggestion
The HTTP request you sent was received by Symfony and the kernel that handled your request.
```

File: src/content/1.7/basics/introduction.md [hunk 15]
```diff
@@ -90,3 +90,60 @@ The driving idea is that we want our code to be more robust, more modular, and f
...
+
+Symfony found the right Back controller, matching your URL, the `PreferencesController`.
+
+The `PreferencesController` returned a HTTP response containing a HTML document rendered by Twig.
```
Comment (kpodemski) [inline src/content/1.7/basics/introduction.md:148]:
```suggestion
The `PreferencesController` returned an HTTP response containing an HTML document rendered by Twig.
```

File: src/content/1.7/basics/introduction.md [hunk 16]
```diff
@@ -90,3 +90,60 @@ The driving idea is that we want our code to be more robust, more modular, and f
...
+
+The Admin controllers use a generic system to choose what Smarty template use. Generic templates to display forms and listings are available and the controller provides the structure configuration (for example it controls what columns and rows are displayed).
+
+### Visit a migrated Back Office controller
+
+When you visit `/admin-{xxx}/index.php/configure/shop/preferences/preferences?_token={yyy}` on your shop you are being returned the Preferences Back Office page.
+
+The HTTP request you sent was received by the Symfony the kernel that handled your request.
+
+Symfony found the right Back controller, matching your URL, the `PreferencesController`.
```
Comment (kpodemski) [inline src/content/1.7/basics/introduction.md:146]:
```suggestion
Symfony found the right Back Office controller, matching your URL, the `PreferencesController`.
```

File: src/content/1.7/basics/introduction.md [hunk 17]
```diff
@@ -90,3 +90,59 @@ The driving idea is that we want our code to be more robust, more modular, and f
 Using a proven and popular open-source framework will allow us to focus on our core business code (managing a cart, handling orders, calculating prices and taxes, generating invoices, etc.) with greater efficiency, while enjoying the stability of a globally recognized framework.
 
 In the documentation, we will refer to the 1.6 framework as the "legacy" framework, as this is a popular [designation](https://en.wikipedia.org/wiki/Legacy_system) used in the software world.
+
+## How it is built inside
+
+Here is a small guide to help you navigate in the software.
+
+Note that a live demo is available on [demo.prestashop.com](https://demo.prestashop.com/).
```
Comment (Progi1984) [inline src/content/1.7/basics/introduction.md:98]: Please don't add link to prestashop.com

File: src/content/1.7/basics/introduction.md [hunk 18]
```diff
@@ -90,3 +90,59 @@ The driving idea is that we want our code to be more robust, more modular, and f
...
+
+### Front office MVC
+
+The Front Office is a small application being powered by PHP and Smarty.
+
+The backend relies on controllers you can find in directory `controllers/front`
+and PrestaShop classes that contain the business logic, mainly from the `classes` folder.
+
+The views and the Javascript come from the installed theme that you will find in `themes/`. You can have multiple themes available on a shop but only one is enabled and in use.
```
Comment (Progi1984) [inline src/content/1.7/basics/introduction.md:107]: What is the behavior with multi shop ?

File: src/content/1.7/basics/introduction.md [hunk 19]
```diff
@@ -90,3 +90,59 @@ The driving idea is that we want our code to be more robust, more modular, and f
...
+
+### Visit a Front controller
+
+When you visit `/` on your shop you are being returned the Homepage.
```
Comment (Progi1984) [inline src/content/1.7/basics/introduction.md:111]:
```suggestion
When you visit `/` on your shop you are being returned to the Homepage.
```

File: src/content/1.7/basics/introduction.md [hunk 20]
```diff
@@ -90,3 +90,59 @@ The driving idea is that we want our code to be more robust, more modular, and f
...
+
+When you visit `/admin{x}/index.php?controller=AdminCarriers&token={y}` on your shop you are being returned to the Carriers Back Office page.
+
+The HTTP request you sent was received by the Back Office _Dispatcher_ which found the right Admin controller (the `AdminCarriersController`).
+
+This `AdminCarriersController` returned an HTTP response containing an HTML document rendered by Smarty.
+
+The Admin controllers use a generic system to choose what Smarty template use. Generic templates to display forms and listings are available and the controller provides the structure configuration (for example it controls what columns and rows are displayed).
+
+### Visit a migrated Back Office controller
```
Comment (Progi1984) [inline src/content/1.7/basics/introduction.md:140]:
```suggestion
### Visit a Symfony Back Office controller
```

File: src/content/1.7/basics/introduction.md [hunk 21]
```diff
@@ -90,3 +90,59 @@ The driving idea is that we want our code to be more robust, more modular, and f
...
+
+When you visit `/admin-{xxx}/index.php/configure/shop/preferences/preferences?_token={yyy}` on your shop you are being returned to the Preferences Back Office page.
```
Comment (Progi1984) [inline src/content/1.7/basics/introduction.md:142]:
```suggestion
When you visit `/admin-{xxx}/index.php/configure/shop/preferences/preferences?_token={yyy}` on your shop, you are being returned to the Preferences Back Office page.
```

File: src/content/1.7/basics/introduction.md [hunk 22]
```diff
@@ -90,3 +90,59 @@ The driving idea is that we want our code to be more robust, more modular, and f
...
+For legacy pages, the views and the Javascript can be found in `admin-dev/themes/default`.
+
+The Symfony backend relies on the controller you can find in `src/PrestaShopBundle/Controller/Admin` and PrestaShop logic that mostly comes from the `src` folder, and also some `classes` files.
+
+For migrated pages, the views can be found in `src/PrestaShopBundle/Resources/views`.
+The Javascript can be found in `admin-dev/themes/new-theme/js`.
+
+### Visit a legacy Back Office controller
+
+When you visit `/admin{x}/index.php?controller=AdminCarriers&token={y}` on your shop you are being returned to the Carriers Back Office page.
```
Comment (Progi1984) [inline src/content/1.7/basics/introduction.md:132]:
```suggestion
When you visit `/admin{x}/index.php?controller=AdminCarriers&token={y}` on your shop, you are being returned to the Carriers Back Office page.
```

**Review by Progi1984 — COMMENTED** [review]
Some feedbacks

**Comment by matks** [comment]
After the introduction of multiple doc versions, rebasing this PR was complex. I rebuilt it in https://github.com/PrestaShop/docs/pull/1119

**Pull request closed without merging**